lib/linked_list.rb

```diff
@@ -18,71 +19,161 @@ def initialize
 
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(1)
     def add_first(value)
-      raise NotImplementedError
-    end
+      if @head.nil?
+        @head = Node.new(value)
+      else 
+        new_nodelet = Node.new(value, @head)
+        new_nodelet.next = @head 
+        @head = new_nodelet 
+      end 

```

Comment:
You could rewrite this as:

```suggestion
    @head = Node.new(value, @head)
```

This way the new node's next points to the old head (which might be `nil`).